src/Pages/ADT3DScenePage/ADT3DScenePage.tsx

```diff
@@ -209,27 +224,38 @@ const ADT3DScenePageBase: React.FC<IADT3DScenePageProps> = ({
         [environmentPickerOptions?.storage, setBlobContainerUrl]
     );
 
-    const handleEnvironmentUrlChange = useCallback(
-        (envUrl: string, envUrls: Array<string>) => {
+    const handleAdtInstanceChange = useCallback(
+        (
+            adtInstance: string | IADTInstance,
+            adtInstances: Array<string | IADTInstance>
+        ) => {
             deeplinkDispatch({
-                type: DeeplinkContextActionType.SET_ADT_URL,
+                type: DeeplinkContextActionType.SET_ADT_INSTANCE,
                 payload: {
-                    url: envUrl
+                    adtInstance
                 }
             });
-            if (environmentPickerOptions?.environment?.onEnvironmentChange) {
-                environmentPickerOptions.environment.onEnvironmentChange(
-                    envUrl,
-                    envUrls
+            if (environmentPickerOptions?.adt?.onAdtInstanceChange) {
+                environmentPickerOptions.adt.onAdtInstanceChange(
+                    adtInstance,
+                    adtInstances
                 );
             }
         },
-        [deeplinkDispatch, environmentPickerOptions?.environment]
+        [deeplinkDispatch, environmentPickerOptions?.adt]
     );
 
     // update the adapter if the ADT instance changes
     useEffect(() => {
         adapter.setAdtHostUrl(deeplinkState.adtUrl);
+        if (environmentPickerOptions?.adt?.selectedItemLocalStorageKey) {
```

Comment:
The intention here is to update the local storage key which the consumer app is using through "`selectedItemLocalStorageKey`". Do you remember that, we had this decision from the meeting "whenever the 3d Scenes app storage key for selected adt url is updated, the ADT Explorer app's selected adt url should also be updated". This is what it does. Since now the setting the local storage with selected values are using the new keys and are not dependent on the passed selectedItemLocalStorageKey prop, we needed to do it manually here. All these optional local storage key props should be deprecated soon, and this local storage update in ADT explorer side should be handled in the consumer side in the future. This is a temporary solution that I put a comment for. 

Does it make sense?

Reply:
I thought this would be handled as part of the migration work you do in the localstorage manager. Why do we need migration in both places?

Reply:
we are not doing migration here, we only update the consumer's selected adt using their choice of local storage key whenever we update ours (which will be deprecated soon as indicated in the comment)